int16 io
---
string csum
